src/main/java/allocator/Allocator.java

```diff
@@ -0,0 +1,39 @@
+package allocator;
+
+import student.Student;
+import studentlist.StudentList;
+import university.University;
+import university.UniversityRepository;
+
+public class Allocator {
+    
+    private StudentList studentList;
+    private UniversityRepository universityList;
+
+    public Allocator(StudentList studentList, UniversityRepository universityList) {
+        this.studentList = new StudentList(studentList);  // don't want to modify the original list
```

Comment:
A new copy of the list can be useful if we do implement a feature in the future that is able to revert any changes. However, if we would want to consistently run multiple allocation rounds, perhaps modifying the original list might be better for memory purposes